pkg/stub/stub.go

```diff
@@ -543,7 +544,7 @@ func (stub *stub) connClosed() {
 		return
 	}
 
-	os.Exit(0)
+	runtime.Goexit() //The exit code can be determined when the coroutine exits
```

Comment:
I'm not sure why a `runtime.Goexit()` would be better here than not doing anything (other than maybe recording the fact that we lost connection) and simply returning instead? If the stub has been started by `stub.Start()` or `stub.Run()` it is sitting in `ttrpc.Server.Serve()` which should return (soon) once we return from here. That should cause `stub.Run()` to return if the caller used it that to start the stub, or otherwise `stub.Wait()` to return if that is used to wait for the stub to stop.

I think one alternative we could do is to
- either record in `connect()` the fact that we've taken a `socketpair` from the env. (IOW. we are running as a pre-connected plugin launched by the runtime)
- and use that stored state to `os.Exit()` here, or check here for the presence of the same env. var and simply use that fact to do the same

Reply:
Or I think something like this could work, too:

```diff
diff --git a/pkg/stub/stub.go b/pkg/stub/stub.go
index 88fd2e1..b68344f 100644
--- a/pkg/stub/stub.go
+++ b/pkg/stub/stub.go
@@ -502,6 +502,15 @@ func (stub *stub) connect() error {
                        return fmt.Errorf("invalid socket (%d) in environment: %w", fd, err)
                }
 
+               userOnClose := stub.onClose
+               stub.onClose = func() {
+                       if userOnClose != nil {
+                               userOnClose()
+                       }
+                       log.Infof(noCtx, "Connection from environment lost, exiting...")
+                       os.Exit(0)
+               }
+
                return nil
        } 
```